pkg/solana/logpoller/types.go

```diff
@@ -0,0 +1,98 @@
+package logpoller
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+	"fmt"
+
+	"github.com/gagliardetto/solana-go"
+)
+
+type PublicKey solana.PublicKey
+
+// Scan implements Scanner for database/sql.
+func (k *PublicKey) Scan(src interface{}) error {
+	return scanFixedLengthArray("PublicKey", solana.PublicKeyLength, src, k[:])
+}
+
+// Value implements valuer for database/sql.
+func (k PublicKey) Value() (driver.Value, error) {
+	return k[:], nil
+}
+
+func (k PublicKey) ToSolana() solana.PublicKey {
+	return solana.PublicKey(k)
+}
+
+type Hash solana.Hash
+
+// Scan implements Scanner for database/sql.
+func (h *Hash) Scan(src interface{}) error {
+	return scanFixedLengthArray("Hash", solana.PublicKeyLength, src, h[:])
+}
+
+// Value implements valuer for database/sql.
+func (h Hash) Value() (driver.Value, error) {
+	return h[:], nil
+}
+
+func (h Hash) ToSolana() solana.Hash {
+	return solana.Hash(h)
+}
+
+type Signature solana.Signature
+
+// Scan implements Scanner for database/sql.
+func (s *Signature) Scan(src interface{}) error {
+	return scanFixedLengthArray("Signature", solana.SignatureLength, src, s[:])
+}
+
+// Value implements valuer for database/sql.
+func (s Signature) Value() (driver.Value, error) {
+	return s[:], nil
+}
+
+func (s Signature) ToSolana() solana.Signature {
+	return solana.Signature(s)
+}
+
+func scanFixedLengthArray(name string, maxLength int, src interface{}, dest []byte) error {
+	srcB, ok := src.([]byte)
+	if !ok {
+		return fmt.Errorf("can't scan %T into %s", src, name)
```

Comment:
Frequent errors like this are good to make as exported errors and wrap them whenever possible.

```
var ErrScanFailure = errors.New("failed to scan")

// ...

return fmt.Errorf("%w %T into %s", ErrScanFailure, src, name)
```

This allows calling functions from external packages to optionally handle errors more gracefully.